File: 2016 maj/c++/2.cpp
#include <iostream>
#include <fstream>
#include <string>
using namespace std;


int main()
{
    ifstream in("dane_6_2.txt");
    ofstream out("wyniki_6_2.txt");

    for (int i = 1; i <= 3000; i++)
    {
        string line;
        getline(in, line);
        
        string s, key;
        int k = 0;
        bool slowo = true;
        for (char c: line)
        {
            if (c == ' ')
            {
                slowo = false;
                continue;
            }
            if (slowo)
                s.push_back(c);
            else
                key.push_back(c);
        }
        if (key.size() > 0)
            k = stoi(key);

        for (char c: s)
            out << char('A' + ((c - 'A' - k) % 26 + 26) % 26);
        out << '\n';
    }
}

File: 2016 maj/python/2.py
with open("dane_6_2.txt", "r") as f:
    file = f.read().strip().split('\n')

def decode(s, key):
    return ''.join([chr(65 + ((ord(i) - 65 - key) % 26 + 26) % 26) for i in s])

with open("wyniki_6_2.txt", "w") as f:
    for s in file:
        s = s.split()
        slowo, key = s[0], 0
        if len(s) > 1:
            key = int(s[1])
        f.write(decode(slowo, key) + '\n')